CHANGELOG.md

```diff
@@ -5,23 +5,73 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
+
+This release reworks the access token API and fully implements extensions to the COSE structures provided by `coset`
+that allow using pre-defined cryptographic backends.
+This way, it is no longer required for applications to implement cryptographic operations and logic to verify that
+COSE structures match the specification.
+
+Note: The long-term plan is to integrate these extensions into `coset` at some point (assuming that they would be
+interested in this code). This might come with minor API adjustments (e.g. by removing some parameters to builder
+methods that already have their own builder method, but whose fields we can't access from outside of `coset`), but the
+basic design of this module should now be quite stable
+Applications can expect to only have to make minor changes in calls to these operations for the forseeable future.
+
+### Added
+
+- An entire new submodule `token/cose` that is responsible for easier handling of creation and verification of COSE
+  structures.
+  Instead of manually defining glue code to cryptographic libraries for each call and manually checking whether the COSE
+  structure is specification compliant, this module takes care of most of the validation and leaves the basic
+  cryptographic operations to reusable predefined cryptographic backends
```

Comment:
```suggestion
  Instead of manually defining glue code to cryptographic libraries for each call and checking whether the COSE
  structure is specification-compliant, this module takes care of most of the validation and leaves the basic
  cryptographic operations to reusable predefined cryptographic backends.
```